### PR TITLE
Product carousel

### PR DESCRIPTION
**Pull request by Crash1212**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ import React from 'react';
 import ReactDOM from 'react-dom';
 import { Provider } from 'react-redux';
 
-import App from './components/app';
+import App from './components/App';
```
Comment (bandvov) [inline src/index.js:5]: please change to app

**Review by bandvov — APPROVED** [review]

File: src/index.js (hunk 1 above)
Comment (Crash1212) [reply src/index.js]: fixed

File: src/pages/product-list-page/product-list-filter/models-filter/models-filter.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        )
+      );
+    } else {
+      dispatch(
+        setModelsFilter([...new Set([...modelsFilter, event.target.name])])
+      );
+    }
+  };
+
+  const [hideFilters, setHideFilters] = useState(true);
```
[the remark this answers is not in the document]
Comment (Crash1212) [reply src/pages/product-list-page/product-list-filter/models-filter/models-filter.js]: fixed

File: src/pages/products-carousel/products-carousel.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
+import React, { useEffect } from 'react';
+import AwesomeSlider from 'react-awesome-slider';
+import { Link } from 'react-router-dom';
+import 'react-awesome-slider/dist/styles.css';
+import { useDispatch, useSelector } from 'react-redux';
+import { useStyles } from './products-carousel.style';
+import { setModelsFilter } from '../../redux/products/products.actions';
+import { getModelsByCategory } from '../../redux/model/model.actions';
+
+const ProductsCorousel = ({ category }) => {
```
[the remark this answers is not in the document]
Comment (Crash1212) [reply src/pages/products-carousel/products-carousel.js:19]: fixed

File: src/pages/product-list-page/product-list-filter/patterns-filter/patterns-filter.js [hunk 4]
```diff
@@ -45,27 +46,37 @@ const PatternsFilter = () => {
...
 
+  const [hideFilters, setHideFilters] = useState(true);
+
   return (
     <FormGroup data-cy='patterns_filter'>
-      <Typography id='patterns' gutterBottom>
-        {PATTERN_TEXT[language].value}:
+      <Typography
+        className={styles.filterName}
+        id='patterns'
```
[the remark this answers is not in the document]
Comment (Crash1212) [reply src/pages/product-list-page/product-list-filter/patterns-filter/patterns-filter.js]: fixed

File: src/pages/product-list-page/product-list-filter/models-filter/models-filter.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+  return (
+    <FormGroup data-cy='category_filter'>
+      <Typography
+        className={styles.filterName}
+        id='categories'
```
[the remark this answers is not in the document]
Comment (Crash1212) [reply src/pages/product-list-page/product-list-filter/models-filter/models-filter.js]: fixed

File: src/redux/model/model.sagas.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
+import { takeEvery, put, call } from 'redux-saga/effects';
+import { push } from 'connected-react-router';
+import { setModels } from './model.actions';
+import getItems from '../../utils/client';
+import { setError } from '../error/error.actions';
+import { GET_MODELS_BY_CATEGORY } from './model.types';
+
+export function* handleModelsLoad({ payload }) {
+  try {
+    console.log(payload);
```
[the remark this answers is not in the document]
Comment (Crash1212) [reply src/redux/model/model.sagas.js]: fixed

File: src/pages/product-list-page/product-list-filter/patterns-filter/patterns-filter.js [hunk 7]
```diff
@@ -45,27 +46,37 @@ const PatternsFilter = () => {
...
-              checked={
-                !!patternsFilter.find((filter) => filter === pattern[1].value)
-              }
-            />
-          }
-          label={pattern[language].value}
-          onChange={handlePatternChange}
-        />
-      ))}
+      {hideFilters ||
```
[the remark this answers is not in the document]
Comment (Crash1212) [reply src/pages/product-list-page/product-list-filter/patterns-filter/patterns-filter.js]: fixed

File: src/pages/product-list-page/product-list-filter/patterns-filter/patterns-filter.js [hunk 8]
```diff
@@ -45,27 +46,37 @@ const PatternsFilter = () => {
     }
   };
 
+  const [hideFilters, setHideFilters] = useState(true);
```
[the remark this answers is not in the document]
Comment (Crash1212) [reply src/pages/product-list-page/product-list-filter/patterns-filter/patterns-filter.js]: fixed

File: src/pages/product-list-page/product-list-filter/models-filter/models-filter.js [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+        gutterBottom
+        onClick={() => setHideFilters(!hideFilters)}
+      >
+        <span>{MODEL_TEXT[language].value}:</span>
+        <span style={{ textDecoration: 'underline' }}>{models.length}</span>
+      </Typography>
+      {hideFilters ||
```
[the remark this answers is not in the document]
Comment (Crash1212) [reply src/pages/product-list-page/product-list-filter/models-filter/models-filter.js]: fixed

File: src/pages/product-list-page/product-list-filter/colors-filter/colors-filter.js [hunk 10]
```diff
@@ -45,27 +46,37 @@ const ColorsFilter = () => {
...
-              checked={
-                !!colorsFilter.find((filter) => filter === color[1].value)
-              }
-            />
-          }
-          label={color[language].value}
-          onChange={handleColorChange}
-        />
-      ))}
+      {hideFilters ||
```
[the remark this answers is not in the document]
Comment (Crash1212) [reply src/pages/product-list-page/product-list-filter/colors-filter/colors-filter.js]: fixed

File: src/pages/product-list-page/product-list-filter/colors-filter/colors-filter.js [hunk 11]
```diff
@@ -45,27 +46,37 @@ const ColorsFilter = () => {
     }
   };
 
+  const [hideFilters, setHideFilters] = useState(true);
```
[the remark this answers is not in the document]
Comment (Crash1212) [reply src/pages/product-list-page/product-list-filter/colors-filter/colors-filter.js]: fixed

**Pull request merged**